checks.d/mysql.py

```diff
@@ -333,27 +673,53 @@ def _collect_system_metrics(self, host, db, tags):
+            # At last, get mysql cpu data out of psutil or procfs
+
             try:
-                # See http://www.kernel.org/doc/man-pages/online/pages/man5/proc.5.html
-                # for meaning: we get 13 & 14: utime and stime, in clock ticks and convert
-                # them with the right sysconf value (SC_CLK_TCK)
-                proc_file = open("/proc/%d/stat" % pid)
-                data = proc_file.readline()
-                proc_file.close()
-                fields = data.split(' ')
-                ucpu = fields[13]
-                kcpu = fields[14]
-                clk_tck = os.sysconf(os.sysconf_names["SC_CLK_TCK"])
-
-                # Convert time to s (number of second of CPU used by mysql)
-                # It's a counter, it will be divided by the period, multiply by 100
-                # to get the percentage of CPU used by mysql over the period
-                self.rate("mysql.performance.user_time",
-                          int((float(ucpu) / float(clk_tck)) * 100), tags=tags)
-                self.rate("mysql.performance.kernel_time",
-                          int((float(kcpu) / float(clk_tck)) * 100), tags=tags)
+                proc = None
+                if PSUTIL_AVAILABLE:
+                    proc = psutil.Process(pid)
+
+                ucpu, scpu, tcpu = None, None, None
+                if hasattr(proc, 'cpu_times') :
+                    ucpu = proc.cpu_times()[0]
+                    scpu = proc.cpu_times()[1]
+
+                elif Platform.is_linux():
+                    # See http://www.kernel.org/doc/man-pages/online/pages/man5/proc.5.html
+                    # for meaning: we get 13 & 14: utime and stime, in clock ticks and convert
+                    # them with the right sysconf value (SC_CLK_TCK)
+                    clk_tck = os.sysconf(os.sysconf_names["SC_CLK_TCK"])
+                    proc_file = open("/proc/%d/stat" % pid)
+
+                    data = proc_file.readline()
+                    proc_file.close()
+                    fields = data.split(' ')
+                    # Convert time to s (number of second of CPU used by mysql)
+                    # It's a counter, it will be divided by the period, multiply by 100
+                    # to get the percentage of CPU used by mysql over the period
+                    #
+                    # FIX(?) : /proc stats are ticks since boot-up, so
+                    # dividing by clk_tck isn't but a conversation to seconds -
+                    # multiplying * 100 does not convert it into a percentage
+                    # Does reporting as a rate make that make sense?
+                    # ucpu = int((float(fields[13]) / float(clk_tck)) * 100)
+                    # scpu = int((float(fields[14]) / float(clk_tck)) * 100)
+                    ucpu = int((float(fields[13]) / float(clk_tck)))
+                    scpu = int((float(fields[14]) / float(clk_tck)))
+
+                elif Platform.is_windows() is not True:
```

Comment:
Nitpick it would be more pythonic to write:
```python
elif not Platform.is_windows()
```

But you can get rid of that. Let's just assume we have psutil